modules/sdk-core/src/bitgo/wallet/wallets.ts

```diff
@@ -538,6 +542,75 @@ export class Wallets implements IWallets {
       .result();
   }
 
+  /**
+   * Re-share wallet with existing spenders of the wallet
+   * @param walletId
+   * @param userPassword
+   */
+  async reshareOfcAccountWithSpenders(walletId: string, userPassword: string): Promise<void> {
+    const wallet = await this.bitgo.coin('ofc').wallets().get({ id: walletId });
+    if (!wallet?._wallet?.enterprise) {
+      throw new Error('Enterprise not found for the wallet');
+    }
+
+    const enterpriseUsersResponse = await this.bitgo
+      .get(`/api/v2/enterprise/${wallet?._wallet?.enterprise}/user`)
+      .result();
+    // create a map of users for easy lookup - we need the user email id to share the wallet
+    const usersMap = new Map(
+      [...enterpriseUsersResponse?.adminUsers, ...enterpriseUsersResponse?.nonAdminUsers].map((obj) => [obj.id, obj])
+    );
+    wallet?._wallet?.users?.forEach(async (user) => {
+      // user should be a spender and not an admin
+      if (user.permissions.includes('spend') && !user.permissions.includes('admin')) {
+        const userObject = usersMap.get(user.user);
+        const shareParams = {
+          walletId: walletId,
+          user: user.user,
+          permissions: user.permissions.join(','),
+          walletPassphrase: userPassword,
+          email: userObject.email.email,
+          coin: wallet.coin,
+          reshare: true,
+        };
+        wallet.shareWallet(shareParams);
+      }
+    });
+  }
+
+  /**
+   * Generate a random password
+   * @param   {Number} numWords     Number of 32-bit words
+   * @returns {String}          base58 random password
+   */
+  generateRandomPassword(numWords = 5): string {
```

Comment:
why are you redefining this? it already exists in the SDK - https://github.com/BitGo/BitGoJS/blob/97c0f3ce5aa83bd4d412ba278be931473bc7c923/modules/sdk-api/src/bitgoAPI.ts#L1106

Reply:
We are aware of this, moving it to a util method and import it at both the places because as of now the sdk-api is not imported in sdk-core (circular dependency issue).